libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -48,13 +48,58 @@ void StaticAnalyzer::endVisit(ContractDefinition const&)
 
 bool StaticAnalyzer::visit(FunctionDefinition const& _function)
 {
+	m_localVarUseCount.clear();
 	m_nonPayablePublic = _function.isPublic() && !_function.isPayable();
 	return true;
 }
 
 void StaticAnalyzer::endVisit(FunctionDefinition const&)
 {
 	m_nonPayablePublic = false;
+	for (auto var = m_localVarUseCount.begin(); var != m_localVarUseCount.end(); ++var)
+	{
+		if (var->second == 0)
+		{
+			warning(var->first->location(), "Unused local variable");
+		}
+	}
+}
+
+bool StaticAnalyzer::visit(Identifier const& _identifier)
+{
+	if (auto var = dynamic_cast<VariableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+	{
+		if (var->isLocalVariable() && var->name() != "")
+		{
+			if (m_localVarUseCount.find(var) != m_localVarUseCount.end())
+			{
+				m_localVarUseCount[var] += 1;
+			}
+		}
+	}
+	return true;
+}
+
+void StaticAnalyzer::endVisit(Identifier const&)
```

Comment:
Please remove this here and in the header file (also endVisit for VariableDeclaration).